### DISCUSSION TITLE
OAuth - Handling the first authorization code

### DISCUSSION THREAD

**Discussion by Jakebluhm** [Jericho/ZoomNet#130] [General]
I am wondering how others are getting the authorization code from when a user downloads the program into their C# Code. Currently to test, I navigate to my Zoom project's 'local test' tab in Zoom Marketplace.

![image](https://user-images.githubusercontent.com/41808367/117676319-6fd11800-b17b-11eb-8376-df0441a0f2d9.png)

There I "install" the app and agree to permissions. Then I am redirected to my "Redirect URL for OAuth" URL that I populated under App Credentials tab. I then copy the authorization code into my C# code, specifically I save it to the accessToken variable. From there ZoomNet works and handles new authorization token and refresh tokens. I am able to access the data I want after that.

How should the authorization code be used from when an end user downloads and agrees to permissions? How can I get this 'authorization code' into my C# code without hard coding it in?

**Comment by Jericho** [comment]
I follow the exact same "manually copy the authorization code to my C# code" flow that you describe. Then again, that's on my dev machine, for development purposes and not in production.

You could probably add a little bit of code in your landing page (AKA your 'redirect URL for OAuth') to make a call to the Zoom API which would force ZoomNet to convert the auth code into the necessary token. Something like retrieving the user's information for example. This would enable you to display "Hello <...user first name goes here...>" on this landing page.
